pkg/nvml/gpm.go

```diff
@@ -57,14 +57,22 @@ type GpmMetricsGetVType struct {
 func (l *library) GpmMetricsGetV(metricsGet *GpmMetricsGetType) GpmMetricsGetVType {
 	return GpmMetricsGetVType{metricsGet.convert()}
 }
+
+// nvmlGpmMetricsGetStub is a stub function that can be overridden for testing.
+var nvmlGpmMetricsGetStub = nvmlGpmMetricsGet
+
 func (metricsGetV GpmMetricsGetVType) V1() Return {
 	metricsGetV.metricsGet.Version = 1
-	return nvmlGpmMetricsGet(metricsGetV.metricsGet)
+	return nvmlGpmMetricsGetStub(metricsGetV.metricsGet)
 }
 
 func (l *library) GpmMetricsGet(metricsGet *GpmMetricsGetType) Return {
 	metricsGet.Version = GPM_METRICS_GET_VERSION
-	return nvmlGpmMetricsGet(metricsGet.convert())
+	nvmlMetricsGet := metricsGet.convert()
+	ret := nvmlGpmMetricsGetStub(nvmlMetricsGet)
+	metricsGet.Metrics = nvmlMetricsGet.Metrics
```

Comment:
@klueska we create a local instance of an nvml type before the function call and we set the return value explicitly here.

Reply:
OK, something changed since I made my previous comment. I was still seeing:
```
func (l *library) GpmMetricsGetV(metricsGet *GpmMetricsGetType) GpmMetricsGetVType {
	return GpmMetricsGetVType{metricsGet.convert()}
}
```
whereby we lose our reference to the original `metricsGet` variable as soon as we do `metricsGet.convert()`. The current code seems to basically do what I was suggesting now.

The one difference being that you do:
```
metricsGet.Metrics = nvmlMetricsGet.Metrics
```
instead of:
```
*metricsGet = nvmlMetricsGet.convert()
```
This is equivalent for now -- but your version would need to be updated if any other fields become OUT fields in the future, wheras using `convert()` avoids this.